### ISSUE TITLE
minor colReorder problem remaining

### ISSUE THREAD

**Issue by gergness** [rstudio/DT#534]
#532 is awesome, thanks so much @shrektan ! I did encounter one remaining bug - it only happens with server set to TRUE, so possibly it should just be mentioned in the documentation.

 To reproduce:

1) Create a DT with colReorder, a filter, and with server set to TRUE
2) Use the filter to filter a column
3) Reorder the column that you've just filtered
4) Try to reset the filter

When you do this, it's clear that the filter isn't being applied to the right column. In the example below, I get a javascript popup about the numeric type vs character type, but if all columns are character, then there's no error but I believe the old filter is applied to the wrong column.

```r
# Same a #531 but this problem is only for server=TRUE
library(shiny)
library(DT)

ui <- basicPage(
  DT::dataTableOutput('foo')
)

server <- shinyServer(function(input, output, session) {
  output$foo = DT::renderDataTable(
    iris, server = TRUE,
    filter = "top", 
    caption = 'colReorder + filter can have trouble',
    extensions = 'ColReorder',
    options = list(colReorder = TRUE)
  )
})
runApp(shinyApp(ui, server))
```
### Initial State
![image](https://user-images.githubusercontent.com/2104579/39325101-f5a91242-4956-11e8-991e-2f291cfc64d6.png)

### Set filter
![image](https://user-images.githubusercontent.com/2104579/39325127-090eb832-4957-11e8-8d9e-18eded13f2b6.png)

### Move filtered column
![image](https://user-images.githubusercontent.com/2104579/39325156-18ad1c16-4957-11e8-8ea7-d54ec4dd8802.png)

### Try to reset filter
![image](https://user-images.githubusercontent.com/2104579/39325174-22b7ca94-4957-11e8-89ba-2a3f117d82c4.png)

**Comment by shrektan** [comment]
I can confirm this. It seems like the server doesn't capture the right column info for sorting. I'll investigate it.

**Comment by shrektan** [comment]
There're two issues, in fact:

1. If one filter is in using and gets reordered (e.g.,  the fifth being moved to the second ), the seach text info sent to the server side will use the new position (the second). This is odd because it's the original column position that should be sent to the server... I think we may be able to fix it if we call the filter event again, whenever the column gets reordered.
1. The sorted position info being sent to the server is always the new position. However, the datatables library doesn't send the col-reorder info to the server... How to let the server knows the current column orders? I don't really know...

Suddenly what @yihui blogged in https://yihui.name/en/2018/01/back-to-dt/ comes to my mind ... 😹 

> The core features are okay, but the extensions… They are my biggest headache, but they do look useful, so users love using them, but are constantly surprised why they don’t work in certain cases. Again, it is due to the combinatorial effect.

**Comment by shrektan** [comment]
I don't think we might be able to fix this but we should document it.

**Comment by yihui** [comment]
It's finally fixed, after 5.5 years... You can test the development version via

```r
remotes::install_github('rstudio/DT')
```
